### ISSUE TITLE
Allows running trace-viewer's code in d8 and node

### ISSUE THREAD

**Issue by nedn** [google/trace-viewer#984]
More context is in https://code.google.com/p/chromium/issues/detail?id=478237

**Comment by dj2** [comment]
So, I have a feeling this is going to be tricky. The original reason we switched to all HTML imports is because the order of loading of scripts and HTML imports isn't well defined.

We had all kinds of problems getting deferred scripts and HTML imports to load in the right order for the examples/trace_viewer.html to actually work.

I'm assuming to do this you're going to convert the HTML importer and models over to JS files again? Have you taken a look at how the loading will work?

**Comment by nedn** [comment]
Hi Dan, I will look at this in more detail in give you the answer. I think our fallback plan can be vulcanize the thing, then strip out the HTML part & run d8 with it.

**Comment by dj2** [comment]
It would be better if we can make it work with HTML and JS files. Having to vulcanize and strip bits out sounds like it could be error prone.

**Comment by natduca** [comment]
I think the vulcanize is actually pretty safe, as long as we pair it with continuous test coverage. Eg if run tests runs all the tests for the things we want working in d8.

I don't want us to move to js files again. If we go that route, then we'd be blocked on es6 imports, which are a ways out so we'd have to do yet another loader... at least with imports, they work in production, and we can vulcanize for d8 support.

Possibly down the road we could hook d8's implementation of load() so that it goes into our own html5 module loader if needed.

**Comment by anniesullie** [comment]
FWIW, I've had to do some pretty complex HTML parsing outside a browser on other projects, and now that there is a HTML5 parsing standard it is not nearly as error-prone as it sounds. We do need to make sure to use an HTML5-compliant parser (instead of assuming we'll have well-formed XHTML or something). I think we could use html5lib + beautiful soup? http://www.crummy.com/software/BeautifulSoup/bs4/doc/#installing-a-parser

**Comment by natduca** [comment]
https://github.com/google/trace-viewer/blob/master/third_party/tvcm/tvcm/parse_html_deps.py is actually using beautifulsoup already. I've not really experimented with html5lib, that sounds interesting. I do know that what we've got seems to mostly work, ~phew~! :)

Moving to html imports was a great thing, it really simplified and reduced our load. So hopefully whatever solution we come up with will keep us on html5 imports for our native code and just wrangle d8 into submission  hehe.

**Comment by natduca** [comment]
Next steps are:
- Finalize the implementation of run_d8_tests
- Fix tests until they're all passing
- Add run_d8_tests to travis
- Remove d8_unittest.py compat tests

When we've finished that, we can actually start thinkging about telemetry<->tracing hookup. :)

**Comment by natduca** [comment]
![image](https://cloud.githubusercontent.com/assets/412396/8498523/b79bbba2-213c-11e5-8c29-e1444b2ed417.png)

**Comment by natduca** [comment]
It works. :D
![image](https://cloud.githubusercontent.com/assets/412396/8498631/ef01a754-213d-11e5-91bd-8a18e511fcd6.png)

**Comment by natduca** [comment]
I believe this is fixed. Ned, want to do the honors?

**Comment by nedn** [comment]
Another chapter in the history of tracing!
Celebration video: https://youtu.be/dQw4w9WgXcQ
